frontend/ui/opdsparser.lua

```diff
@@ -79,6 +79,9 @@ function OPDSParser:parse(text)
     -- but will kick the ass of luxl
     text = text:gsub("<br>", "<br />")
     text = text:gsub("<br/>", "<br />")
+    -- Same deal with hr
+    text = text:gsub("<hr>", "<hr />")
+    text = text:gsub("<hr/>", "<hr />")
```

Comment:
That's necessary for this luxl thing? `<hr/>` is perfectly valid. The main reason `<hr />` is still practiced is compatibility with some late '90s/early 2000s browsers and either people don't know that and they're stupidly copying the way it's always been done or perhaps they just prefer the look of `<hr />`.

Btw, while I imagine [the full list](http://xahlee.info/js/html5_non-closing_tag.html) is unlikely to show up, what about `<img/>`?

Reply:
I don't have anything with img tags to check, but at the very list it's rarely (if ever) going to be empty, which might not trip the state machine the same way?

Reply:
Depends if the person writing the code wants to send technically valid HTML when JS is disabled. Of course `<img src="">` or `<img/>` is invalid, but if the src is going to be filled out by JS (doesn't matter one bit if we're talking vanilla, jQuery, React, Angular, etc., etc.) then `<img src="placeholder.jpg" alt="placeholder">` might be technically valid but obviously in a completely meaningless way.

Fancier still is something like: `<img src="data:," alt="placeholder">`. Valid per spec, no HTTP request, and the validator will give it a nice green checkmark, but any person who thinks it's therefore oh-so-much better than just `<img src="">` lacks a brain. Of course if the validator is part of your process you want to avoid false positives, no argument there.

But anyway, even if it's not done on purpose I'm sure it can happen by accident. ;-) Like if you're using React to do something like `<img {src}>` where `src` is accidentally `null` instead of `src="image.jpg"`. Sounds like I made up some pretty stupid code but I'm sure there's way worse out there in the wild.

Reply:
PS `<wbr>` is always used just like that, in any case.

Reply:
I dunno, that all seems fairly unlikely ^^.